actions/remote.go

```diff
@@ -0,0 +1,168 @@
+package actions
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/qri-io/dag"
+	"github.com/qri-io/dag/dsync"
+	"github.com/qri-io/qri/p2p"
+	"github.com/qri-io/qri/repo"
+	"github.com/qri-io/qri/repo/profile"
+)
+
+// DsyncStartPush posts the dataset's dag.Info to the remote to get a sessionID and diff
```

Comment:
Just cracked this PR for review, read this comment. Only feeling I have is "we might finally be able to reason about dsync"